### DISCUSSION TITLE
Feature Request For Creation/Edit records in Modals from Filament Resource Tables

### DISCUSSION THREAD

**Discussion by MACscr** [filamentphp/filament#1412] [Ideas]
I love the way that relation tables use modals for editing records and creating records. This is super useful and doesn't require really any work on the devs part as they can even reuse their main resources forms/tables for it. As discussed in discord, I would like this same functionality available for the main resource ListRecords page as well. With a simple trait, we should be able to specify that we want records to be created/opened in modals versus a whole new page. This is important for simple records that have few form inputs and of course dont have a relation manager tied to them, thus no need for its own creation/edit page. This would work from both the Create action button on the ListRecords page and also the edit/view table row action buttons as well.

Discussed on discord:
https://discord.com/channels/883083792112300104/936768710620880947/937283288404471828

**Comment by danharrin** [comment]
https://github.com/laravel-filament/filament/issues/1415#issuecomment-1030958200

> #1478 - from `v2.10.0` you will be able to add the following traits to your List page, and delete the dedicated Edit and Create pages:
> 
> ```
> use ListRecords\Concerns\CanCreateRecords;
> use ListRecords\Concerns\CanDeleteRecords;
> use ListRecords\Concerns\CanEditRecords;
> ```
> 
> Alternatively, you can create new resources using this structure with:
> 
> ```shell
> php artisan make:filament-resource Blog --simple
> ```
>
> This will replace the List page with a "Manage" page instead with these modals enabled by default.
